server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -161,6 +161,17 @@ public boolean parsesArrayValue() {
         return false;
     }
 
+    /**
+     * Whether this mapper can handle an object value during document parsing. If true,
+     * when an object is encountered during parsing, the document parser will pass the
+     * whole object to the mapper. If false, the object name becomes part of the dotted
+     * field name of each internal value when subobjects property is set to false,
+     * otherwise it gets mapped to an ObjectMapper.
+     */
```

Comment:
One aspect that is not entirely clear to me reading this comment is: do we need to consult this method when subobjects is set to true? The comment would make me think so, but the only reason why we need the method is for subobjects false / auto - flattening? 

I would think that when subobjects is true, there are two situations:
1) there is a field mapped that corresponds to the object we are parsing, which we provide to the mapper, no matter if it supports objects or not. It it does not, parsing will throw error
2) there is no field mapped for the object we are parsing, in which case we will create a new object mapper dynamically (assuming dynamic:true)

Does it add any value to consult this method when subobjects is true?

We could clarify, unless I got this wrong, that we call this method only when subobjects is false, and we use it to discriminate between field mappers that should be provided with the object to be parsed (situation 1. above), and field mappers that need to continue parsing and map nothing (situation 2. above, no object mapper created).

Reply:
Could we also rename the method to supportsParsingObject ?